torch/fx/experimental/symbolic_shapes.py

```diff
@@ -532,13 +540,19 @@ def create_symbol(self, val: int, *, sname: str) -> "sympy.Expr":
 
         # Create a duck sized int if necessary
         if val not in self.val_to_var:
-            sympy_expr = sympy.Symbol(f"s{len(self.var_to_val)}", positive=True, integer=True)
+            global COUNTER
+            sympy_expr = Symbol(f"s{COUNTER}", positive=True, integer=True)
+            COUNTER += 1
+            sympy_expr.shape_env = self
```

Comment:
I need to fix this up: the problem is that Sympy deduplicates symbols that have the same name, and this is very confusing when there are multiple ShapeEnvs over different generations. Need to figure out proper way to impede this.

Reply:
Can we just initialize a sympy with a starting int? Sympy then increments this as it works, and we read it out as we wrap up the frame.

Then, when you have "generation"-al systems like dynamo, the lifecycle is:

- Enter frame
- Make shape_env(counter_pos)
- do stuff (shape_env increments counter_pos as it goes)
- Exit frame, record counter_pos + 1 at shape_env end time
- Enter frame
- Make shape_env(counter_pos)
...
etc

Reply:
My main problem is that I want the number to reset every fresh frame, because I generally want my compilation to be indifferent to what other computation went on. So I can't have a generation counter either.

Reply:
Sorry, I think I misunderstood you, what is the desired behavior?

Reply:
I get to number from 0 every iteration, but I have a distinct Symbol.